orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/artifacts/FindArtifactsFromResourceTask.java

```diff
@@ -0,0 +1,103 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.artifacts;
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.kork.artifacts.model.ExpectedArtifact;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.clouddriver.model.Manifest;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import com.netflix.spinnaker.orca.pipeline.util.ArtifactResolver;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+public class FindArtifactsFromResourceTask extends AbstractCloudProviderAwareTask implements Task {
+  public static final String TASK_NAME = "findArtifactsFromResource";
+
+  @Autowired
+  ArtifactResolver artifactResolver;
+
+  @Autowired
+  OortService oortService;
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    StageData stageData = stage.mapTo(StageData.class);
+    String account = getCredentials(stage);
+    Set<String> expectedArtifactIds = new HashSet<>(stageData.expectedArtifactIds);
+    List<ExpectedArtifact> expectedArtifacts = artifactResolver.getExpectedArtifacts(stage);
+    Map<String, Object> outputs = new HashMap<>();
+
+    List<ExpectedArtifact> unresolvedExpectedArtifacts = expectedArtifacts.stream()
+      .filter(a -> expectedArtifactIds.contains(a.getId()))
+      .filter(a -> a.getBoundArtifact() == null
+        || (stageData.preferDefaultArtifacts == false
+            && a.getBoundArtifact().equals(a.getDefaultArtifact())))
+      .collect(Collectors.toList());
+    Set<String> unresolvedExpectedArtifactNames = unresolvedExpectedArtifacts.stream()
+      .map(a -> a.getMatchArtifact().getName())
+      .collect(Collectors.toSet());
+
+    Manifest manifest = oortService.getManifest(account, stageData.location,
+      stageData.manifestName);
+
+    if (manifest != null) {
+      List<Artifact> artifactsFromManifest = manifest.getArtifacts().stream()
+        .filter(a -> unresolvedExpectedArtifactNames.contains(a.getName()))
+        .collect(Collectors.toList());
+      List<ExpectedArtifact> resolvedExpectedArtifacts = new ArrayList<>();
+
+      for (ExpectedArtifact expectedArtifact : unresolvedExpectedArtifacts) {
+        for (Artifact newArtifact : artifactsFromManifest) {
```

Comment:
I think we have logic for this already: https://github.com/spinnaker/orca/pull/1914/files#diff-ab3dd9a63d59c02eb0cf7191ce664c72R124

Reply:
Done.